### PR TITLE
PC backports sentinel3: geometry cleanup

### PR DESCRIPTION
**Pull request by ircwaves**
**Description:**
This builds on #15 and adds the common geometry repair strategies from PC to stactools:
- ensure counter-clockwise winding 
- handle antimeridian crossings
- round precision of geometric values to 4 decimal places

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [ ] Documentation has been updated to reflect changes, if applicable.
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

**Comment by gadomski** [comment]
FYI @ircwaves I'm going to hold off on a review here until #16 lands, so I can see a clean diff of this PR's changes.

File: setup.cfg
Comment (gadomski) [inline setup.cfg:53]:
```suggestion
    antimeridian >= 0.2.6
```

**Review by gadomski — CHANGES_REQUESTED** [review]

**Review by gadomski — APPROVED** [review]

File: setup.cfg
Comment (ircwaves) [reply setup.cfg]: Thanks...just cribbed the mspc version.

**Pull request merged**
